### PR TITLE
BUGFIX: support switching between target types

### PR DESCRIPTION
**Pull request by inteon**
Missed in https://github.com/cert-manager/trust-manager/pull/193

On master, switching from the ConfigMap target type to the Secret target type was not supported correctly (or vice-versa).
I added additional tests and fixed this bug in this PR.

### CODE REVIEW

File: pkg/bundle/bundle.go [hunk 1]
```diff
@@ -179,6 +190,16 @@ func (b *bundle) reconcileBundle(ctx context.Context, req ctrl.Request) (result
 
 	// Find all desired targetResources.
 	{
+		var targetTypeValue targetType
+		switch {
+		case bundle.Spec.Target.Secret != nil:
+			targetTypeValue = secretTarget
+		case bundle.Spec.Target.ConfigMap != nil:
+			targetTypeValue = configMapTarget
+		default:
```
[the remark this answers is not in the document]
Comment (inteon) [reply pkg/bundle/bundle.go]: Thanks for letting me know, I also added a test where both targets are created.

**Review by SgtCoDFish — APPROVED** [review]
/lgtm
/approve

Thanks! 😁

**Comment by Jiawei0227** [comment]
I remember I explicitly add validation webhook to prevent switch target types just for ease of maintainance of the code :) Thanks for adding the support tho

**Pull request merged**